include/extractor/guidance/toolkit.hpp

```diff
@@ -350,7 +350,8 @@ inline bool requiresNameAnnounced(const std::string &from, const std::string &to
         const auto ref_begin = name.find_first_of('(');
         if (ref_begin != std::string::npos)
         {
-            out_name = name.substr(0, ref_begin);
+            //we might need to trim a space, if there is a reference
+            out_name = name.substr(0, ref_begin - (ref_begin > 0 && name[ref_begin-1] == ' '));
```

Comment:
By the way, the stdlib's string type is horribly broken for such things (it pre-dates the stdlib included in the standard), that's why there is string_ref / string_view or hand-made equivalents e.g. in the Chromium code base.

- https://groups.google.com/a/chromium.org/forum/#!topic/chromium-dev/EUqoIz2iFU4
- https://news.ycombinator.com/item?id=8704318